### ISSUE TITLE
invalid linker name in argument '-fuse-ld=gold'

### ISSUE THREAD

**Issue by NatoBoram** [ipfs/go-ipfs#5063]
#### Type:

Build failed on Android arm64

#### Description:

```bash
# github.com/ipfs/go-ipfs/cmd/ipfs
/data/data/com.termux/files/usr/lib/go/pkg/tool/android_arm64/link: running aarch64-linux-android-clang failed: exit status 1
clang-6.0: error: invalid linker name in argument '-fuse-ld=gold'
make: *** [Rules.mk:118: install_unsupported] Error 2
```

#### Video
[ZIP](https://github.com/ipfs/go-ipfs/files/2061429/build.zip) | [IPFS](https://ipfs.io/ipfs/QmbvEgddHqEefusYMJQ2c417mwtYE19jquAwpkXJpiHKez)

**Comment by Stebalien** [comment]
I'm sorry, I'm just as lost as you. :disappointed:

**Comment by NatoBoram** [comment]
Hm. This is all what I've found : https://en.wikipedia.org/wiki/Gold_(linker)

> In software engineering, **gold** is a [linker](https://en.wikipedia.org/wiki/Linker_(computing)) for [ELF](https://en.wikipedia.org/wiki/Executable_and_Linkable_Format) files. It became an official GNU package and was added to [binutils](https://en.wikipedia.org/wiki/Binutils) in March, 2008 and first released in binutils version 2.19. Gold was developed by Ian Lance Taylor and a small team at Google. The motivation for writing gold was to make a linker that is faster than the [GNU linker](https://en.wikipedia.org/wiki/GNU_linker), especially for large applications coded in C++.
>
> Unlike the GNU linker, gold doesn't use the [BFD library](https://en.wikipedia.org/wiki/Binary_File_Descriptor_library) to process object files. While this limits the object file formats it can process to ELF only, it is also claimed to result in a cleaner and faster implementation without an additional abstraction layer. The author cited complete removal of BFD as a reason to create a new linker from scratch rather than incrementally improve the GNU linker.
>
> To specify gold in a makefile, one sets the LD or LD environmental variable to ld.gold. To specify gold through a compiler option, one can use the gcc option `-fuse-ld=gold`.

And I'm still as lost as lost can be. Maybe a different linker is needed in Android?

**Comment by Stebalien** [comment]
Hm. Actually, try installing `binutils-gold`.

**Comment by NatoBoram** [comment]
Oh, I think it works! Where's the output for `make install_unsupported`?

**Comment by Stebalien** [comment]
I'd rather not do one-off builds. We really should make Android a supported platform so I'd like to get this working with the NDK/SDK and automated.

Note: that hostname issue should be fixed in go 1.11.

**Comment by Stebalien** [comment]
Some instructions that may help: https://www.sajalkayan.com/post/go-android-binary.html
